crates/solidity/inputs/schema/grammar/05-expressions/03-numbers/productions.yml

```
@@ -2,43 +2,78 @@
 
 - name: "NumericLiteral"
   kind: "Parser"
-  unversioned:
-    sequence:
-      - choice:
-          - reference: "HexLiteral"
-          - reference: "DecimalLiteral"
-      - optional:
-          reference: "NumberUnit"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - reference: "HexLiteral"
+            - reference: "DecimalLiteral"
+        - optional:
+            reference: "NumberUnit"
+    0.5.0:
+      sequence:
+        - reference: "DecimalLiteral"
+        - optional:
+            reference: "NumberUnit"
 
 - name: "HexLiteral"
   kind: "Scanner"
-  unversioned:
-    sequence:
-      - terminal: "0x"
-      - oneOrMore:
-          reference: "HexCharacter"
-      - zeroOrMore:
-          sequence:
-            - terminal: "_"
-            - oneOrMore:
-                reference: "HexCharacter"
+  versioned:
+    0.4.11:
+      sequence:
+        - choice:
+            - terminal: "0x"
+            - terminal: "0X"
```

nit, just for readability in spec:
```suggestion
        - terminal: "0"
        - choice:
            - terminal: "x"
            - terminal: "X"
```

---

donert